File: src/main/java/no/novari/flyt/discovery/service/model/dtos/IntegrationMetadataDto.java
package no.novari.flyt.discovery.service.model.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

@Getter
@Builder
@EqualsAndHashCode
@Jacksonized
public class IntegrationMetadataDto {

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private final Long id;

    @NotNull
    private final Long sourceApplicationId;

    @NotBlank
    private final String sourceApplicationIntegrationId;

    private final String sourceApplicationIntegrationUri;

    @NotBlank
    private final String integrationDisplayName;

    @NotNull
    private final Long version;

    @Valid
    private final InstanceMetadataContentDto instanceMetadata;

}

File: src/main/kotlin/no/novari/flyt/discovery/service/model/dtos/IntegrationMetadataDto.kt
package no.novari.flyt.discovery.service.model.dtos

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class IntegrationMetadataDto(
    @field:JsonProperty(access = JsonProperty.Access.READ_ONLY)
    val id: Long? = null,
    @field:NotNull
    var sourceApplicationId: Long? = null,
    @field:NotBlank
    val sourceApplicationIntegrationId: String? = null,
    val sourceApplicationIntegrationUri: String? = null,
    @field:NotBlank
    val integrationDisplayName: String? = null,
    @field:NotNull
    var version: Long? = null,
    @field:Valid
    val instanceMetadata: InstanceMetadataContentDto? = null,
)
